polylogue/document_store.py

```
@@ -256,79 +289,87 @@ def persist_document(
-                    "title": polylogue.get("title") or title,
-                    "slug": polylogue.get("slug") or slug,
-                    "provider": provider,
-                    "conversationId": conversation_id,
+    # Protect user edits: if file is dirty and we're not explicitly allowing overwrite
+    if provider and conversation_id and existing_dirty and state_entry:
+        if force and not allow_dirty:
+            # --force alone not enough to overwrite edited files
+            raise ValueError(
+                f"File {markdown_path} has local edits. "
+                "Use --allow-dirty with --force to overwrite, or omit --force to preserve edits."
+            )
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)